lib/capability-tool.js

```diff
@@ -0,0 +1,280 @@
+  * @param {String} [params.resourceType] - Resource type (e.g. 'Patient', 'Observation'), optional.
+  * @param {Object} params.capabilityType - Capability type (e.g. 'interaction', 'searchParam').
+  * @param {Object} [params.where] - Check additional level of compatibility, optional.
+  * @param {String} [params.where.code] - Specify a code (e.g., 'read' for interaction), optional.
+  * @param {String} [params.where.name] - Specify a name (e.g., 'type' for searchParam), optional.
+  *
+  * @return {Boolean} If the capability, code, or name is supported for the resource.
+  */
+  supportFor({ resourceType, capabilityType, where } = {}) {
+    let capabilities;
+    if (resourceType) {
+      capabilities = this.resourceCapabilities({ resourceType });
+    } else {
+      capabilities = this.serverCapabilities();
+    }
+
+    if (!capabilities) { return false; }
+
+    const capability = capabilities[capabilityType];
+
+    if (where && capability) {
+      const whereParam = Object.keys(where)[0];
+      const serverCapability = capability.find(capabilityItem => (
+        capabilityItem[whereParam] === where[whereParam]
+      ));
+      return serverCapability !== undefined;
+    }
+
+    return capability !== undefined;
+  }
+
+  /**
+  * Return interactions available for a resource type based on the capability statement.
+  *
+  * @example
+  *
+  * const capabilities = new CapabilityTool(capabilityStatement);
+  * const supportedPatientInteractions = capabilities.interactionsFor({ resourceType: 'Patient'});
+  * console.log(supportedPatientInteractions);
+  *
+  * @param {Object} params - The capability parameters.
+  * @param {String} params.resourceType - Resource type (e.g. 'Patient', 'Observation').
+  *
+  * @return {Array} A list of supported interactions for the given resource type.
+  */
+  interactionsFor({ resourceType } = {}) {
+    const resourceCapabilities = this.resourceCapabilities({ resourceType });
+    if (resourceCapabilities === undefined) { return false; }
+    return resourceCapabilities.interaction.map(interaction => (interaction.code));
+  }
+
+  /**
+  * Return searchParams available for a resource type based on the capability statement.
+  *
+  * @example
+  *
+  * const capabilities = new CapabilityTool(capabilityStatement);
+  * const supportedPatientSearchParams = capabilities.searchParamsFor({ resourceType: 'Patient'});
+  * console.log(supportedPatientSearchParams);
+  *
+  * @param {Object} params - The capability parameters.
+  * @param {String} params.resourceType - Resource type (e.g. 'Patient', 'Observation').
+  *
+  * @return {Array} A list of supported searchParams for the given resource type.
+  */
+  searchParamsFor({ resourceType } = {}) {
+    const resourceCapabilities = this.resourceCapabilities({ resourceType });
+    if (resourceCapabilities === undefined) { return false; }
+    return resourceCapabilities.searchParam.map(param => (param.name));
```

Comment:
`searchParam` is not required, so this should handle the case `searchParam` is `undefined`.